src/System.IO.Compression/tests/Performance/Perf.DeflateStream.cs

```diff
@@ -10,75 +10,70 @@ namespace System.IO.Compression.Tests
 {
     public class Perf_DeflateStream
     {
-        private static List<object[]> _compressedFiles;
-        private static List<object[]> _byteArraysToCompress;
-
-        /// <summary>
-        /// Yields an Enumerable list of paths to GZTestData files
-        /// </summary>
-        public static IEnumerable<object[]> CompressedFiles()
+        /// <returns></returns>
+        private static string CreateCompressedFile(CompressionType type)
         {
-            if (_compressedFiles == null)
-            {
-                PerfUtils utils = new PerfUtils();
-                _compressedFiles = new List<object[]>();
-                // Crypto random data
-                byte[] bytes = new byte[100000000];
-                var rand = RandomNumberGenerator.Create();
-                rand.GetBytes(bytes);
-                string filePath = utils.GetTestFilePath() + ".gz";
-                using (FileStream output = File.Create(filePath))
-                using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
-                    zip.Write(bytes, 0, bytes.Length);
-                _compressedFiles.Add(new object[] { filePath });
-
-                // Create a compressed file with repeated segments
-                bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(100000));
-                filePath = utils.GetTestFilePath() + ".gz";
-                using (FileStream output = File.Create(filePath))
-                using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
-                    for (int i = 0; i < 1000; i++)
+            byte[] bytes;
+            const int fileSize = 100000000;
+            PerfUtils utils = new PerfUtils();
+            string filePath = utils.GetTestFilePath() + ".gz";
+            switch (type)
+            {    
+                case CompressionType.CryptoRandom:
+                    bytes = new byte[fileSize];
+                    var rand = RandomNumberGenerator.Create();
+                    rand.GetBytes(bytes);
+                    using (FileStream output = File.Create(filePath))
+                    using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
+                        zip.Write(bytes, 0, bytes.Length);
+                    break;
+                case CompressionType.RepeatedSegments:
+                    bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(fileSize / 1000));
+                    using (FileStream output = File.Create(filePath))
+                    using (GZipStream zip = new GZipStream(output, CompressionMode.Compress))
+                        for (int i = 0; i < 1000; i++)
```

Comment:
Good idea!

We initiate zlib with a windowBits of -15 which zlib interprets as a memory window size of of (2^15 - 262) = 32506 bytes. That should give enough leeway when writing 1000, but I'll add a test for 100 bytes at a time to see those results as well.